File: xyz/Euler/2/src/Solution.cpp
#include <iostream>

using namespace std;

long process(long sentinel) {
    long sum = 0;

    long prev = 0;
    long curr = 2;

    long temp;
    while (curr < sentinel) {
        sum += curr;
        temp = prev + (curr << 2);
        prev = curr;
        curr = temp;
    }

    return sum;
}

int main() {
    int testCases;

    long sentinel;

    while (cin >> testCases) {
        for (int i = 0; i < testCases; i++) {
            cin >> sentinel;
            cout << process(sentinel) << endl;
        }
    }
}

File: xyz/Euler/2/src/Solution.go
package main

import (
    "fmt"
    "io"
)

func process(sentinel int64) int64 {
    sum := int64(0)

    prev := int64(0)
    curr := int64(2)

    for curr < sentinel {
        sum += curr
        prev, curr = curr, prev + (curr << 2)
    }

    return sum
}

func main() {
    var testCases int

    var sentinel int64

    for {
        _, err := fmt.Scan(&testCases)
        if err == io.EOF {
            break
        }
        for i := 0; i < testCases; i++ {
            fmt.Scan(&sentinel)
            fmt.Println(process(sentinel))
        }
    }
}
